pkg/codegen/docs/gen_creation_example_for_yaml.go

```diff
@@ -0,0 +1,193 @@
+package docs
+
+import (
+	"bytes"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/pulumi/pulumi/pkg/v3/codegen"
+	"github.com/pulumi/pulumi/pkg/v3/codegen/schema"
+)
+
+// collapseToken converts an exact token to a token more suitable for
```

Comment:
Slightly surprised we don't have this defined somewhere else we can borrow from?  Feels like we should have a standard procedure for this "simplified type token" we maintain one place.

Reply:
This is because this simplified form of tokens only applies to YAML programs and I _borrowed_ it from YAML program-gen. Tokens of the shape `package:member` (implicit module) are technically not valid when binding a program or schema